### ISSUE TITLE
Feature: Chat

### ISSUE THREAD

**Issue by brettwilcox** [GRVYDEV/Project-Lightspeed#5]
### Chat and Communication

Can we build in real time communication using NodeBB? I'm thinking if you can find a way to sync up the time of the comments and the time of the live stream, you could keep everything in sync with something like Influx.

- https://nodebb.org

**Comment by GRVYDEV** [comment]
So it looks like nodebb costs money and I have implemented a realtime chat before. https://github.com/GRVYDEV/realtime-chat-server and https://github.com/GRVYDEV/realtime-chat-client so I would look into doing this through our own solution. Ideally we would use elixir since it is IMO the best tech for doing real-time chat however I decided to forgo that for now since it is a bit more complex to deploy

**Comment by brettwilcox** [comment]
NodeBB is GPL3 and completely OSS.

- https://github.com/NodeBB/NodeBB
- https://community.nodebb.org
- https://docs.nodebb.org/api/

My concern is ability to scale, API, and moderation abilities. Think 20,000+ users.

**Comment by GRVYDEV** [comment]
> NodeBB is GPL3 and completely OSS.
> 
>     * https://github.com/NodeBB/NodeBB
> 
>     * https://community.nodebb.org
> 
>     * https://docs.nodebb.org/api/
> 
> 
> My concern is ability to scale, API, and moderation abilities. Think 20,000+ users.

Elixir phoenix can handle 2mil+ websocket connections on 1 server so scale wont be an issue :)

**Comment by GRVYDEV** [comment]
Also checkout this repo https://github.com/Glimesh/glimesh.tv They are using elixir for a full streaming site frontend and have implemented moderation and bans

**Comment by brettwilcox** [comment]
> Elixir phoenix can handle 2mil+ websocket connections on 1 server so scale wont be an issue :)

**Comment by brettwilcox** [comment]
> Elixir phoenix can handle 2mil+ websocket connections on 1 server so scale wont be an issue :)

I agree! I was more referring to a mature project with a full fledge API for integration. I am going to likely use NodeBB and perform an integration with https://github.com/tsugiproject/tsugi but I will be interested in seeing what you can come up with.

**Comment by brettwilcox** [comment]
https://matrix.org - This would be interesting as well...
